MDNetworking/MDReplicator.cs

```diff
@@ -82,6 +110,43 @@ public void RegisterReplication(Node Instance)
         if (NodeMembers.Count > 0)
         {
             NodeList.Add(new ReplicatedNode(Instance, NodeMembers));
+            List<object> networkIdUpdates = new List<object>();
+            foreach (MDReplicatedMember member in NodeMembers)
+            {
+                KeyToMemberMap.Add(member.GetUniqueKey(), member);
+                if (MDStatics.IsServer())
+                {
+                    if (!NetworkIdKeyMap.ContainsKey(member.GetUniqueKey()))
+                    {
+                        uint networkid = GetReplicationId();
+                        NetworkIdKeyMap.AddNetworkKeyIdPair(networkid, member.GetUniqueKey());
+                        NetworkIdKeyMap.CheckBuffer(networkid, member);
+                        networkIdUpdates.Add(networkid);
+                        networkIdUpdates.Add(member.GetUniqueKey());
+                    }
+                }
+            }
+            if (networkIdUpdates.Count > 0)
+            {
+                Rpc(nameof(UpdateNetworkIdMap), networkIdUpdates);
+            }
+        }
+    }
+
+    /// Process our settings
+    private void ProcessSettingsForMember(MDReplicatedMember ReplicatedMember, MDReplicatedSetting[] Settings)
+    {
+        foreach (MDReplicatedSetting setting in Settings)
+        {
+            switch ((Settings)setting.Key)
```

Comment:
I think this will be an issue with different enums being used as the key, as the first enum value will always be zero.
We might have to do something like this and make the key an int
```cs
public class MDReplicator
{
    public enum Settings
    {
        ProcessWhilePaused = 0,
        GroupName,
        ReplicatedMemberType,
        Max
    }
}

public class MDClockedReplicatedMember : MDReplicatedMember
{
    public enum Settings
    {
        OnValueChangedEvent = MDReplicator.Settings.Max,
        Max
    }
}

public class CustomClass
{
    public enum Settings
    {
        CustomSetting = MDClockedReplicatedMember.Settings.Max
    }
}
```

What's annoying with that is we'll have to track the Max chaining and know which one is highest...

Reply:
IS this really a problem if you look in the replicator where I call it I do this

```csharp
ProcessSettingsForMember(NodeMember, ParseParameters(typeof(Settings), Settings));
```

And if you look at ParseParameters it looks like this,
```csharp
public static MDReplicatedSetting[] ParseParameters(Type TypeToLookFor, MDReplicatedSetting[] Parameters)
    {
        List<MDReplicatedSetting> SettingsList = new List<MDReplicatedSetting>();
        foreach (MDReplicatedSetting setting in Parameters)
        {
            if (setting.Key.GetType() == TypeToLookFor)
            {
                SettingsList.Add(setting);
            }
        }
        return SettingsList.ToArray();
    }
```

So in short I expect people to filter by the enum type before they process the settings, in which case I don't think this is a problem?

Reply:
Ah okay, yeah that should be okay